rfcs/0013-synthea-future-claims-and-automation.md

```diff
@@ -0,0 +1,148 @@
+
+Large Synthetic Data Generation:
+  - Each PI (Program Increment), approximately every 8 weeks, 10 - 60 million Synthea beneficiaries will be generated and available for on-demand performance and load testing on TEST and PROD SBX databases. The number of beneficiaries in the dataset size is meant to reflect the size and shape of production. The process will be similar to that of automated recurring generation, however the memory and computational power of the AWS instance will be larger for time and cost effectiveness. 
+  
+  - Using a m5a.24xlarge EC2 instance, it took 6 hours to generate 1 million synthetic beneficiaries. If one extrapolates from this benchmark, it would take 60 hours or a little under 3 days to generate 10 million, and 360 hours, or 15 days to generate 60 million beneficiaries.
+
+  - For the time being, a single reusable instance will be used. Pipelining multiple EC2 instances to generate data more efficiently is something to explore at a later point, as well as running the generation and load testing of large data manually every PI. 
+
+Required BFD Application Changes:
+When CCW data is ingested by the BFD pipeline, the application has filter logic for beneficiary UPDATEs with reference years prior to the current year by default. However, Synthea data must be able to UPDATE prior years. Additionally, future claims that are staged in S3 will need to be automatically ingested by the pipeline. For both Synthea and CCW data to be ingested by the pipeline application, the back-dated beneficiary filtering will have to be turned off for Synthea data, and a timestamp for future claims needs to be recognized. 
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+Automated Generation & Load Plan:
+* On a quarterly basis a cron job within Jenkins will be set up to start up and SSH into a provisioned AWS EC2 instance that will be turned off when not used, to run an Ansible script that takes a batch size and randomly generated year between 1 and 5 and will execute other scripts that:
+  - Run currently manual steps in the `Synthea Test Plan` such as accessing and comparing the latest Synthea end-properties file, running queries in PROD SBX to determine database ranges for beneficiary and claim properities, and ensuring the next batch of data will not overlap with existing data ranges.
+  - Generate batches of both synthetic current and future data
+  - Load into AWS S3, and when applicable, trigger the BFD pipeline to transform and load synthetic data into database for TEST, PROD SBX, and PROD.
+
+* On a weekly basis, future claims will be loaded into the database depending on their timestamp.
```

Comment:
This is explained further down in the required changes section, but to avoid ambiguity Im taking away that section and putting it with this section to talk about detailed future claims as a whole.